File: LeetCode/python3/263.py
class Solution:
    def isUgly(self, num: int) -> bool:
        if num < 1:
            return False
        while num % 2 == 0:
            num //= 2
        while num % 3 == 0:
            num //= 3
        while num % 5 == 0:
            num //= 5
        return True if num == 1 else False

File: LeetCode/java/263.java
class Solution {
    public boolean isUgly(int num) {
        if (num == 1)
            return true;
        while (true) {
            if (num == 2 || num == 3 || num == 5)
                return true;
            int ori = num;
            if (ori % 2 == 0)
                ori /= 2;
            else if (ori % 3 == 0)
                ori /= 3;
            else if (ori % 5 == 0)
                ori /= 5;
            if (ori == num)
                return false;
            num = ori;
        }
    }
}